### PR TITLE
Simulation Stack - November release

### PR DESCRIPTION
**Pull request by ferdonline**
### New package versions:
 - neurodamus-core 2.8.0 (Support for Sonata nodes)
 - MVDTool 2.2.1 (Improved Sonata reading) and 2.3.0 (Unified API reading Sonata / Mvd3+mecombo.tsv)
 - SynapseTool 0.5.5 (Sonata translation fix)
 - libsonata 0.1.2 (Several improvements, \_\_len\_\_ and allowing std::move; docs)
 - HighFive 2.1 (patch 1) (See [Release Notes](https://github.com/BlueBrain/HighFive/releases/tag/v2.1))

### Recipe fixes:
 - Fixed PYTHONPATH in Neuron and MVDTool
 - Neurodamus depending on py-mvdtool for Sonata 
 - Dropped +python in bare mvdtool package. User should use py-mvdtool
 - spykfunc to build on py-mvdtool~mpi (and dropped old versions)
 - Added Neuron 7.8 (not default)
 - HighFive builds with +Boost by default, and tests for @develop

### CODE REVIEW

**Comment by ferdonline** [comment]
Can we change this to Draft? - It seems we can't.

File: var/spack/repos/builtin/packages/neurodamus-model/package.py
Comment (matz-e) [inline var/spack/repos/builtin/packages/neurodamus-model/package.py]:
```suggestion
    variant('mvdtool',     default=False, description="Enable MVDTool reader for Sonata Nodes")
```

**Review by matz-e — CHANGES_REQUESTED** [review]

**Review by pramodk — COMMENTED** [review]
@fernando: close this and reopen (with different branch?) as new Draft PR to avoid merge.

**Comment by ferdonline** [comment]
With a change request review it cant be merged

**Review by matz-e — DISMISSED** [review]
Haπer now.

File: var/spack/repos/builtin/packages/highfive/package.py
Comment (ferdonline) [inline var/spack/repos/builtin/packages/highfive/package.py]: @alkino why dont we use the tag here?

Comment (alkino) [reply var/spack/repos/builtin/packages/highfive/package.py]: I follow the rest of the file.

Comment (ferdonline) [reply var/spack/repos/builtin/packages/highfive/package.py]: I see... @pramodk any advantage/disadvantage of using release tarballs?

Comment (alkino) [reply var/spack/repos/builtin/packages/highfive/package.py]: maybe we should wait and wait for the patch of parallel hdf5

Comment (ferdonline) [reply var/spack/repos/builtin/packages/highfive/package.py]: did you find problems with parallel hdf5? last time I tested it was fine

Comment (alkino) [reply var/spack/repos/builtin/packages/highfive/package.py]: https://github.com/BlueBrain/HighFive/pull/242

Comment (ferdonline) [reply var/spack/repos/builtin/packages/highfive/package.py]: See the comment. If it would work as it was before let's keep 2.1. Otherwise we need a new release :(

**Comment by ferdonline** [comment]
We will need the merge and tag of https://bbpcode.epfl.ch/code/#/c/47014/

**Comment by ferdonline** [comment]
@pramodk How shall we do for Neuron? Is there the urge to deploy 7.8? There isn't still a tag including the fix required for neurodamus-py, right?

File: deploy/packages/parallel-libraries.yaml
Comment (ferdonline) [inline deploy/packages/parallel-libraries.yaml:38]: Just to explain the change dropping the version.If we specify here and Spack has a different default version then, in case it's used for dependencies too, two versions end up being installed without need. I feel that in these cases we should take the default version, which comes either in the dependency spec, the package recipe, or packages.yaml.

**Comment by pramodk** [comment]
> @pramodk How shall we do for Neuron? Is there the urge to deploy 7.8? There isn't still a tag including the fix required for neurodamus-py, right?

Yeah..there is no fix. Bit considering tight schedule for Yale visit during next few days, we can postpone 7.8 deployment as it will be changing results. We can do that once we return back.

File: deploy/packages/parallel-libraries.yaml
Comment (pramodk) [reply deploy/packages/parallel-libraries.yaml:38]: this looks fine.

File: var/spack/repos/builtin/packages/neurodamus-model/package.py
Comment (matz-e) [inline var/spack/repos/builtin/packages/neurodamus-model/package.py]: MVD = nodes!

Comment (matz-e) [reply var/spack/repos/builtin/packages/neurodamus-model/package.py]: Also, this is not just for SONATA?

File: var/spack/repos/builtin/packages/neurodamus-model/package.py
Comment (matz-e) [inline var/spack/repos/builtin/packages/neurodamus-model/package.py]: If we have an `mpi` variant, this should be `+mpi`!

File: var/spack/repos/builtin/packages/py-mvdtool/package.py
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-mvdtool/package.py:21]: This may break Spykfunc. Could you fix all `py-mvdtool` dependencies to be `py-mvdtool~mpi` unless required otherwise?

**Review by matz-e — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-mvdtool/package.py
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-mvdtool/package.py:38]: Not sure if these decorators stack the right way? Seems like that's triggering the failure. Maybe swapping them would help? If not, I think the `mpi` test should be moved inside the function.

**Review by matz-e — CHANGES_REQUESTED** [review]

**Comment by matz-e** [comment]
Retest this please

File: var/spack/repos/builtin/packages/py-mvdtool/package.py
Comment (ferdonline) [reply var/spack/repos/builtin/packages/py-mvdtool/package.py:38]: Indeed we cannot stack them :( The other way around gives MethodNotFound

**Comment by ferdonline** [comment]
@pramodk @matz-e I think it's ready for review 🎉

File: var/spack/repos/builtin/packages/emsim/package.py
Comment (pramodk) [inline var/spack/repos/builtin/packages/emsim/package.py]: in that case I expect that to be only build dependency?

File: var/spack/repos/builtin/packages/highfive/package.py
Comment (pramodk) [inline var/spack/repos/builtin/packages/highfive/package.py]:
```suggestion
    version('develop', branch='master')
```

File: var/spack/repos/builtin/packages/highfive/package.py
Comment (pramodk) [inline var/spack/repos/builtin/packages/highfive/package.py]:
```suggestion
    version('2.1.1', tag='v2.1.1')
```

File: var/spack/repos/builtin/packages/mvdtool/package.py
Comment (pramodk) [inline var/spack/repos/builtin/packages/mvdtool/package.py]: remove ?

File: var/spack/repos/builtin/packages/neurodamus-model/package.py
Comment (pramodk) [inline var/spack/repos/builtin/packages/neurodamus-model/package.py]: this is more for developer / debugging. Is that necessary?

File: var/spack/repos/builtin/packages/neuron/package.py
Comment (pramodk) [inline var/spack/repos/builtin/packages/neuron/package.py:285]: we always call `set_python_path` from `setup_environment`. What `@when` decorators does here?

File: var/spack/repos/builtin/packages/py-mvdtool/package.py
Comment (pramodk) [inline var/spack/repos/builtin/packages/py-mvdtool/package.py:19]: if we are already deploying 2.3.0, is 2.2.1 required?

**Review by pramodk — COMMENTED** [review]
take a quick look and have a couple of comments/questions.

File: var/spack/repos/builtin/packages/neuron/package.py
Comment (ferdonline) [reply var/spack/repos/builtin/packages/neuron/package.py:285]: the decorator changes the original func. So even if we call it directly we call the "no-op" function

File: var/spack/repos/builtin/packages/py-mvdtool/package.py
Comment (ferdonline) [reply var/spack/repos/builtin/packages/py-mvdtool/package.py:19]: 2.3.0 is a big jump (TSV, refactoring, moves...), so it is nice to make avail the latest 2.2.1 (which ATM has the stuff required for neurodamus)
We can drop the older ones if you want some cleanup

File: var/spack/repos/builtin/packages/neurodamus-model/package.py
Comment (ferdonline) [reply var/spack/repos/builtin/packages/neurodamus-model/package.py]: that is more for developer --> yes, especially myself to avoid I blindly build the wrong develop :) It was also to test the patch to spack core related to clean=False and tags

File: var/spack/repos/builtin/packages/mvdtool/package.py
Comment (ferdonline) [reply var/spack/repos/builtin/packages/mvdtool/package.py]: Sure. Dont approve any PR if people try to reimplement this 😆

File: var/spack/repos/builtin/packages/neuron/package.py
Comment (pramodk) [reply var/spack/repos/builtin/packages/neuron/package.py:285]: Ok understood

File: var/spack/repos/builtin/packages/py-mvdtool/package.py
Comment (pramodk) [reply var/spack/repos/builtin/packages/py-mvdtool/package.py:19]: If 2.3.0 is sufficient and if no one really required 2.2., I would remove it.

File: var/spack/repos/builtin/packages/neurodamus-model/package.py
Comment (pramodk) [reply var/spack/repos/builtin/packages/neurodamus-model/package.py]: If we are just using `tty.warn`, I would be in favour of removing it

**Review by matz-e — APPROVED** [review]

**Pull request merged**
